modules/platforms/dotnet/Apache.Ignite.Core/Common/IgniteProductVersion.cs

```diff
@@ -0,0 +1,216 @@
+
+        /** Revision hash. */
+        private readonly byte[] _revHash;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="major">Major version number.</param>
+        /// <param name="minor">Minor version number.</param>
+        /// <param name="maintenance">Maintenance version number.</param>
+        /// <param name="revTs">Revision date.</param>
+        /// <param name="revHash">Revision hash.</param>
+        public IgniteProductVersion(byte major, byte minor, byte maintenance, DateTime revTs, byte[] revHash)
+            : this(major, minor, maintenance, "", revTs, revHash)
+        {
+            // No-op.
+        }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="major">Major version number.</param>
+        /// <param name="minor">Minor version number.</param>
+        /// <param name="maintenance">Maintenance version number.</param>
+        /// <param name="stage">Stage of development.</param>
+        /// <param name="releaseDate">Revision date.</param>
+        /// <param name="revHash">Revision hash.</param>
+        public IgniteProductVersion(byte major, byte minor, byte maintenance, string stage, DateTime releaseDate, byte[] revHash)
+        {
+            if (revHash != null && revHash.Length != 20)
+            {
+                throw new ArgumentException("Invalid length for SHA1 hash (must be 20): " + revHash.Length);
+            }
+
+            _major = major;
+            _minor = minor;
+            _maintenance = maintenance;
+            _stage = stage;
+            _releaseDate = releaseDate;
+            _revHash = revHash;
+        }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="reader"><see cref="IBinaryRawReader"/></param>
+        public IgniteProductVersion(IBinaryRawReader reader)
+        {
+            IgniteArgumentCheck.NotNull(reader, "reader");
+
+            _major = reader.ReadByte();
+            _minor = reader.ReadByte();
+            _maintenance = reader.ReadByte();
+            _stage = reader.ReadString();
+            _releaseDate = BinaryUtils.JavaTicksToDateTime(reader.ReadLong());
+            _revHash = reader.ReadByteArray();
+        }
+
+        /// <summary>
+        /// Gets the major version number.
+        /// </summary>
+        public byte Major
+        {
+            get { return _major; }
+        }
+
+        /// <summary>
+        /// Gets the minor version number.
+        /// </summary>
+        public byte Minor
+        {
+            get { return _minor; }
+        }
+
+        /// <summary>
+        /// Gets the maintenance version number.
+        /// </summary>
+        public byte Maintenance
+        {
+            get { return _maintenance; }
+        }
+
+        /// <summary>
+        /// Gets the stage of development.
+        /// </summary>
+        public string Stage
```

Comment:
Please add a test for this property.
It always returns `SNAPSHOT` for test builds.

Reply:
Okay, but we already have `TestBinaryConstructor` method that uses `_defaultVersion `with populated Stage property under the hood.